### PR TITLE
`prefer-string-slice`: remove unsafe autofix for `String#substr()`

### LINKED ISSUE #2291: `prefer-string-slice` bad autofix from `.substr()`
```js
let string = "::";

// Prefer `String#slice()` over `String#substr()`.
// eslint(unicorn/prefer-string-slice)
let eslintError = string.substr(-2, 2);
                  ^^^^^^^^^^^^^^^^^^^^
// After applying fix:
let fixed = string.slice(-2, -2 + 2);

eslintError // "::"
fixed // ""

// Uncaught AssertionError [ERR_ASSERTION]: '::' == ''
assert.equal(eslintError, fixed)
```

### PR DESCRIPTION
**Pull request by axetroy**
Fix #2291

When the second argument `length` is equal to the length of the string,

it will autofix to `.slice(start, 0)`

Now we make to `.slice(start, 0 || undefined)` to align the behavior of `.substr(start, length)`

### CODE REVIEW

File: rules/prefer-string-slice.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ function * fixSubstrArguments({node, fixer, context, abort}) {
 	if (argumentNodes.every(node => isNumber(node, scope))) {
 		const firstArgumentText = getParenthesizedText(firstArgument, sourceCode);
 
-		yield fixer.insertTextBeforeRange(secondArgumentRange, `${firstArgumentText} + `);
+		yield fixer.insertTextBeforeRange(secondArgumentRange, `(${firstArgumentText} + `);
+		yield fixer.insertTextAfterRange(secondArgumentRange, ') || undefined');
```
Comment (fisker) [inline rules/prefer-string-slice.js:71]: The original `Math.min()` already ugly, let's turn this fix into suggestion, I don't think people will use the fixed code directly.

Comment (axetroy) [reply rules/prefer-string-slice.js:71]: I admit, this is ugly, but I can't think of a more elegant fix

Comment (fisker) [reply rules/prefer-string-slice.js:71]: > let's turn this fix into suggestion

Use suggestion instead.

Comment (axetroy) [reply rules/prefer-string-slice.js:71]: The source code is too complicated, and I don't have the confidence to refactor it. Close PR

**Comment by fisker** [comment]
Let's just remove the autofix not much value to provide a suggestion.

**Review by fisker — APPROVED** [review]

**Pull request merged**